docs/_posts/2015-10-22-client-implementation-example-session.adoc

```diff
@@ -0,0 +1,289 @@
+---
+layout: page
+title: "Client Implementation Example Session"
+category: dev
+date: 2015-10-22 18:26:21
+order: 6
+---
+:icons: font
+
+This page takes a look at a typical RVI client session, where the client connects to the RVI node, gets notified of an available update, downloads and installs the update, and reports the success back to the server. We'll break this down in detail, and look at each of the steps of the process. You can also download link:/rvi_sota_server/example-client-session.log[the debug log file from this session].
+
+. Registering services
+.. Client connects to RVI node.
+.. RVI node lists the services it has available.
+.. Client registers its available services with the RVI node.
+.. Client gets the fully qualified names of its services from the node in return.
+. Package notification and download
+.. Client receives a notification from the server that a package update is available.
+.. Client tells the server it can start sending the package.
+.. Server tells the client the package's checksum, and the total number of 64kb chunks in the package, and the client acknowledges.
+.. Server starts sending chunks, and the client acknowledges each one.
+.. Once the client has acknowledged all the chunks, the server tells the client the transfer is finished.
+. Package installation and report
+.. The client assembles the chunks, and calls the local software loading manager to install the package.
+.. Once the SLM terminates, the client sends a report back to the server with the success or failure of the package install.
+
+== 1. Registering Services
+
+When the client connects to the RVI node, the first thing that needs to happen is an exchange of information about services: the client registers with the RVI node all of the services it implements, and the RVI node tells the client which services it has available. The RVI node also gives the client a unique name for each service it registers.
+
+Here is how the RVI node notifies the client about an available service:
+
+[source,json]
+----
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "services_available",
+  "params": {
+    "services": [
+      "genivi.org/backend/sota/start"
+    ]
+  }
+}
+----
+
+TIP: Currently, the services the RVI node will make available are `genivi.org/backend/sota/start`, `genivi.org/backend/sota/ack`, `genivi.org/backend/sota/report`, and `genivi.org/backend/sota/packages`.
+
+Here is how the client registers a service with the RVI node:
+
+[source,json]
+----
+{
+  "jsonrpc": "2.0",
+  "id": 2649007917144,
+  "method": "register_service",
+  "params": {
+    "network_address": "sota_client_1:8090",
```

Comment:
I'm just going to add an explanatory note, since these are pulled directly from the log file, which is available to download.